lib/services/local/users.go

```diff
@@ -675,8 +683,24 @@ func (s *IdentityService) DeleteUser(ctx context.Context, user string) error {
 	// each user has multiple related entries in the backend,
 	// so use DeleteRange to make sure we get them all
 	startKey := backend.ExactKey(webPrefix, usersPrefix, user)
-	err = s.DeleteRange(ctx, startKey, backend.RangeEnd(startKey))
-	return trace.Wrap(err)
+	if err = s.DeleteRange(ctx, startKey, backend.RangeEnd(startKey)); err != nil {
+		return trace.Wrap(err)
+	}
+
+	// Delete notification objects associated with this user.
+	if (s.notificationsSvc != NotificationsService{}) {
+		// Delete all user-specific notifications for this user.
+		if err = s.notificationsSvc.DeleteAllUserNotificationsForUser(ctx, user); err != nil {
+			return trace.Wrap(err)
```

Comment:
If we get an error here should we still try to delete the user's notification states?